erdjs/src/apiProvider.ts

```diff
@@ -0,0 +1,68 @@
+import axios from "axios";
+import { IApiProvider } from "./interface";
+import * as errors from "./errors";
+import { Logger } from "./logger";
+import { NetworkStake } from "./networkStake";
+const JSONbig = require("json-bigint");
+
+/*
+ * This is a temporary change, this will be the only provider used, ProxyProvider will be deprecated
+ */
+export class ApiProvider implements IApiProvider {
+  private url: string;
+  private timeoutLimit: number;
+
+  /**
+   * Creates a new ApiProvider.
+   * @param url the URL of the Elrond Api
+   * @param timeout the timeout for any request-response, in milliseconds
+   */
+  constructor(url: string, timeout?: number) {
+    this.url = url;
+    this.timeoutLimit = timeout || 1000;
+  }
+
+  /**
+   * Fetches the Network Stake.
+   */
+  async getNetworkStake(): Promise<NetworkStake> {
+    let response = await this.doGet(`stake`);
+    let payload = response;
+    return NetworkStake.fromHttpResponse(payload);
+  }
+
+  private async doGet(resourceUrl: string): Promise<any> {
+    try {
+      let url = `${this.url}/${resourceUrl}`;
+      let response = await axios.get(url, { timeout: this.timeoutLimit });
+      let payload = response.data;
+
+      return payload;
+    } catch (error) {
+      if (!error.response) {
```

Comment:
Maybe we can refactor this error handling into a function for more readability inside the `doGet`.